File: packages/react-native/android/src/main/java/com/breezliquidsdk/BreezLiquidSDKLogStream.kt
package com.breezliquidsdk

import breez_liquid_sdk.LogEntry
import breez_liquid_sdk.Logger
import com.facebook.react.modules.core.DeviceEventManagerModule.RCTDeviceEventEmitter

class BreezLiquidSDKLogger(
    private val emitter: RCTDeviceEventEmitter,
) : Logger {
    companion object {
        var emitterName = "breezLiquidSdkLog"
    }

    override fun log(l: LogEntry) {
        emitter.emit(emitterName, readableMapOf(l))
    }
}

File: packages/react-native/ios/BreezLiquidSDKLogStream.swift
import Foundation
import BreezLiquidSDK

class BreezLiquidSDKLogger: Logger {
    static let emitterName: String = "breezLiquidSdkLog"

    func log(l: LogEntry) {
        if RNBreezLiquidSDK.hasListeners {
            RNBreezLiquidSDK.emitter.sendEvent(withName: BreezLiquidSDKLogger.emitterName,
                                         body: BreezLiquidSDKMapper.dictionaryOf(logEntry: l))
        }
    }
}